src/libpmemfile-posix/block_array.c

```diff
@@ -256,6 +257,9 @@ block_list_remove(struct pmemfile_vinode *vinode,
 		if (vinode->first_block == moving_block)
 			vinode->first_block = block;
 		relocate_block(block, moving_block);
+		ctree_remove_unlocked(vinode->blocks, block->offset, 1);
+		ctree_insert_unlocked(vinode->blocks, block->offset,
+		    (uint64_t)block);
```

Comment:
It would be easier to understand if ctree_remove would be moved before relocate_block and use moving_block->offset instead of block->offset. Other than that LGTM.